hack/lib/etcd.sh

```diff
@@ -45,13 +45,20 @@ kube::etcd::validate() {
     exit 1
   fi
 
+  # need set the env of "ETCD_UNSUPPORTED_ARCH" on unstable arch.
+  arch=$(uname -m)
```

Comment:
Instead of this you can use `go env GOHOSTARCH` or `go env GOARCH` that both will report the value `ETCD_UNSUPPORTED_ARCH` wants.

This is done for all arches except amd64 (including other arches supported by etcd) when building the docker image.

https://github.com/kubernetes/kubernetes/blob/706cde51c512d8d07e35fc815af4439108a587f9/cluster/images/etcd/Makefile#L138

I also this you can just place `ETCD_UNSUPPORTED_ARCH="$(go env GOHOSTARCH)"` before each call to etcd, or alternatively add the export to the top of this file.

```sh
export ETCD_UNSUPPORTED_ARCH="$(go env GOHOSTARCH)" 
```

Reply:
Good to know, will fix this soon.

Reply:
Although I agree that your solution is good, but seems like kind's doesn't have go installed, so I will revert it back.
```
/home/prow/go/src/k8s.io/kubernetes/hack/lib/etcd.sh: line 23: go: command not found
!!! [0224 06:19:34] Call tree:
!!! [0224 06:19:34]  1: /home/prow/go/src/k8s.io/kubernetes/hack/lib/init.sh:56 source(...)
!!! [0224 06:19:34]  2: ./hack/ginkgo-e2e.sh:26 source(...)
!!! Error in /home/prow/go/src/k8s.io/kubernetes/hack/lib/etcd.sh:23
  Error in /home/prow/go/src/k8s.io/kubernetes/hack/lib/etcd.sh:23. 'ETCD_UNSUPPORTED_ARCH="$(go env GOHOSTARCH)"' exited with status 127
```

Reply:
cross build failed with the same issue as well, 
https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/88399/pull-kubernetes-cross/1231823687401869315/#1:build-log.txt%3A66

Do you know why?

Reply:
Ahh, sorry for the confusion.

Looks like golang is missing from the image used: https://github.com/kubernetes/test-infra/blob/master/images/bootstrap/Dockerfile

Reply:
how about revert this back?

Reply:
let's see if infra team okay with this: https://github.com/kubernetes/test-infra/pull/16447

Reply:
I don't think this is the right approach. Unless I'm misremembering we don't require go on the host, we have a dockerized build (and bazel!)

uname is posix.

Reply:
when we cross build we use the dockerized build, we expect go to be in that container, but not on the host.